src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraRteEditPageTest.kt

```diff
@@ -0,0 +1,174 @@
+                return user
+            }
+
+            override fun remember(memories: Collection<User>) {
+                throw Exception("not implemented")
+            }
+        }
+
+        JiraCoreFormula.Builder()
+            .build()
+            .provision()
+            .use { jira ->
+                DockerisedChrome().start().use { browser ->
+                    val webJira = WebJira(
+                        browser.driver,
+                        jira.getUri(),
+                        user.password
+                    )
+                    val logInAction = scenario.getLogInAction(
+                        webJira,
+                        actionMeter,
+                        userMemory
+                    )
+                    val actions = scenario.getActions(
+                        webJira,
+                        SeededRandom(123),
+                        actionMeter
+                    )
+
+                    logInAction.run()
+                    actions.forEach { action ->
+                        action.run()
+                    }
+                }
+            }
+
+        val results = metrics.map { metric ->
+            metric.result
+        }
+        Assertions.assertThat(results).containsOnly(ActionResult.OK)
+    }
+}
+
+class JiraEditScenario constructor() : Scenario {
+    private lateinit var issueKeyMemory: IssueKeyMemory
+
+    private constructor(issueKeyMemory: IssueKeyMemory?) : this() {
+        issueKeyMemory?.let { this.issueKeyMemory = it }
+    }
+
+    override fun getActions(jira: WebJira, seededRandom: SeededRandom, meter: ActionMeter): List<Action> {
+        initializeIssueKeyMemory(seededRandom)
+        val projectMemory = AdaptiveProjectMemory(random = seededRandom)
+        val jqlMemory = AdaptiveJqlMemory(seededRandom)
+        val issueMemory = AdaptiveIssueMemory(issueKeyMemory, seededRandom)
+
+        val viewDashboard = ViewDashboardAction(
+            jira = jira,
+            meter = meter
+        )
+        val createIssue = CreateIssueAction(
+            jira = jira,
+            meter = meter,
+            seededRandom = seededRandom,
+            projectMemory = projectMemory
+        )
+        val searchWithJql = SearchJqlAction(
+            jira = jira,
+            meter = meter,
+            jqlMemory = jqlMemory,
+            issueKeyMemory = issueKeyMemory
+        )
+        val viewIssue = ViewIssueAction(
+            jira = jira,
+            meter = meter,
+            issueKeyMemory = issueKeyMemory,
+            issueMemory = issueMemory,
+            jqlMemory = jqlMemory
+        )
+        val editIssue = EditIssueAction(
+            jira = jira,
+            meter = meter,
+            issueMemory = issueMemory
+        )
+        val addComment = AddCommentAction(
+            jira = jira,
+            meter = meter,
+            issueMemory = issueMemory
+        )
+
+        val scenario: MutableList<Action> = mutableListOf(
```

Comment:
I like how you narrowed down the action list from `JiraCoreScenario` in order to leave only:
* the RTE actions themselves (`createIssue`, `editIssue`, `addComment`)
* the exploratory actions for `issueMemory` and `issueKeyMemory` (`searchWithJql`, `viewIssue`)

However, it seems we don't need `viewDashboard`.

Reply:
okay, got it